chocolatey-community/chocolatey-packages
# (1password*) Split out package to major versions and add support for 8.x

## Description

The work in this repository splits out the 1password package into multiple packages that are specific for a major version of 1password.

It additionally also adds support for 1password 8.x in a restrictive manner.

## Motivation and Context

To allow specific major versions being installed with ease, as wel as adding support for 1password 8.x.

## How Has this Been Tested?

Create versions of 4.x, 7.x and 8.x of the 1password package.
This can be done in with the following commands:

1. `.\update_all.ps1 '1password' -ForcedPackages '1password\OPW4'`
2. `.\update_all.ps1 '1password*' -ForcedPackages '1password\OPW7 1password7'`
3. `.\update_all.ps '1password'`

You should have a version of 1password4, 1password7 and 1password 8, and their equivalent versions on the 1password package.

Testing should be done in the chocolatey enviromnent

1. Copy all of the created packages to the test environment.
2. Install the `1password` package for version 4.x that you created.
3. Ensure that the `1password4` package was pulled in and installed successfully.
4. Upgrade the `1password` package to the created 7.x version.
5. Ensure that the `1password7` package was pulled in and installed successfully.
6. Upgrade the `1password` package to the created 8.x version.
7. Ensure that the `1password8` package was pulled in and installed successfully.
8. Uninstall the `1password4` package.
9. Ensure the software was successfully uninstalled, and only uninstalled the 4.x version.
10. Uninstall the `1password7` package.
11. Ensure that the software was successfully uninstalled, and only uninstalled the 7.x version.
12. Uninstall the `1password` while using the `-x` argument.
13. Ensure that the software is completely removed now using the auto uninstaller.

## Screenshot (if appropriate, usually isn't needed):

## Types of changes


## Checklist:


- [] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] The changes only affect a single package (not including meta package). (_Multiple packages is touched, but all are related to eachother_)

fixes #1773

https://app.clickup.com/t/20540031/ENGTASKS-3993
